sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/JoinOptimizationSuite.scala

```diff
@@ -92,4 +92,30 @@ class JoinOrderSuite extends PlanTest {
 
     comparePlans(optimized, analysis.EliminateSubqueryAliases(correctAnswer))
   }
+
+  test("broadcasthint sets relation statistics to smallest value") {
+    val input = LocalRelation('key.int, 'value.string)
+
+    val query =
+      Project(Seq($"x.key", $"y.key"),
+        Join(
+          SubqueryAlias("x", input),
+          BroadcastHint(SubqueryAlias("y", input)), Inner, None)).analyze
+
+    val optimized = Optimize.execute(query)
+
+    val expected =
+      Project(Seq($"x.key", $"y.key"),
+        Join(
+          Project(Seq($"x.key"), SubqueryAlias("x", input)),
+          Project(Seq($"y.key"),
+            BroadcastHint(SubqueryAlias("y", input))),
+          Inner, None)).analyze
+
+    comparePlans(optimized, expected)
+
+    assert(optimized.collect {
+      case b @ BroadcastHint(_) if b.statistics.sizeInBytes == 1 => 1
```

Comment:
This is a bit of a nit, but I think what you really want to test here is:

```scala
val broadcastChildren = optimized.collect {
  case Join(_, CanBroadcast(r), _, _) => r
}

assert(broadcastChildren == 1)
```

With the current test something could break in `Project` (for example) that would prevent the broadcast from actually happening.